src/sidebar/store/modules/test/groups-test.js

```diff
@@ -108,4 +139,140 @@ describe('sidebar.store.modules.groups', () => {
       assert.equal(store.focusedGroupId(), privateGroup.id);
     });
   });
+
+  describe('getFeaturedGroups', () => {
+    [
+      {
+        description:
+          'If logged in and there is a restricted group of which the user is not a member show it in `Featured Groups`.',
+        isLoggedIn: true,
+        filteredGroups: [restrictedGroup],
+        expectedGroups: [restrictedGroup],
+      },
+      {
+        description:
+          'If logged in and there is an open group show it in `Featured Groups`.',
+        isLoggedIn: true,
+        filteredGroups: [openGroup],
+        expectedGroups: [openGroup],
+      },
+      {
+        description:
+          'If logged in and there are only groups with membership do not show `Featured Groups`.',
+        isLoggedIn: true,
+        filteredGroups: [publicGroup],
+        expectedGroups: [],
+      },
+      {
+        description:
+          'If logged out and there is a restricted group scoped to the page show it in `Featured Groups`.',
+        isLoggedIn: false,
+        filteredGroups: [restrictedGroup],
+        expectedGroups: [restrictedGroup],
+      },
+      {
+        description:
+          'If logged out and there is an open group show it in `Featured Groups`.',
+        isLoggedIn: false,
+        filteredGroups: [openGroup],
+        expectedGroups: [openGroup],
+      },
+    ].forEach(({ description, isLoggedIn, filteredGroups, expectedGroups }) => {
+      it(description, function() {
+        store.getState().session = { userid: isLoggedIn ? '1234' : null };
+        store.loadGroups(filteredGroups);
+
+        const featuredGroups = store.getFeaturedGroups();
+        const currentlyViewing = store.getCurrentlyViewingGroups();
+
+        assert.isTrue(featuredGroups.length === expectedGroups.length);
+        let i;
```

Comment:
A couple of thoughts here...

* You can declare your local var in the for loop itself, e.g. `(for let i = 0; ...`
* It feels like an equality assertion may be more appropriate than `isTrue` with an equality test
* It may be clearer to compare objects instead of their `id` properties as they should indeed be the same reference
* You could ditch the `for` loop if you wanted, and map the asserts over the Array

Altogether now, you could maybe do something like:

```
featuredGroups.map((fGroup, idx) => assert.deepEqual(fGroup, expectedGroups[idx]);
```

(These thoughts go for the other similar tests in this module)